stacktrace_ext_test.go

```diff
@@ -84,33 +85,35 @@ func TestStacktraceFiltersZapMarshal(t *testing.T) {
 }
 
 func TestStacktraceFiltersVendorZap(t *testing.T) {
-	// We need to simulate a zap as a vendor library, so we're going to create a fake GOPATH
-	// and run the above test which will contain zap in the vendor directory.
+	// We already have the dependencies downloaded so this should be
+	// instant.
+	deps := downloadDependencies(t)
+
+	// We need to simulate a zap as a vendor library, so we're going to
+	// create a fake GOPATH and run the above test which will contain zap
+	// in the vendor directory.
 	withGoPath(t, func(goPath string) {
-		curDir, err := os.Getwd()
+		zapDir, err := os.Getwd()
 		require.NoError(t, err, "Failed to get current directory")
 
 		testDir := filepath.Join(goPath, "src/go.uber.org/zap_test/")
 		vendorDir := filepath.Join(testDir, "vendor")
 		require.NoError(t, os.MkdirAll(testDir, 0777), "Failed to create source director")
 
 		curFile := getSelfFilename(t)
-		//copyFile(t, curFile, filepath.Join(testDir, curFile))
 		setupSymlink(t, curFile, filepath.Join(testDir, curFile))
 
 		// Set up symlinks for zap, and for any test dependencies.
-		setupSymlink(t, curDir, filepath.Join(vendorDir, "go.uber.org/zap"))
-		for _, testDep := range []string{"github.com/stretchr/testify"} {
-			target := filepath.Join(curDir, "vendor", testDep)
-			_, err := os.Stat(target)
-			require.NoError(t, err, "Required dependency (%v) not installed in vendor", target)
-			setupSymlink(t, target, filepath.Join(vendorDir, testDep))
+		setupSymlink(t, zapDir, filepath.Join(vendorDir, "go.uber.org/zap"))
+		for _, dep := range deps {
+			setupSymlink(t, dep.Dir, filepath.Join(vendorDir, dep.ImportPath))
 		}
 
-		// Now run the above test which ensures we filter out zap stacktraces, but this time
-		// zap is in a vendor
+		// Now run the above test which ensures we filter out zap
+		// stacktraces, but this time zap is in a vendor
 		cmd := exec.Command("go", "test", "-v", "-run", "TestStacktraceFiltersZap")
 		cmd.Dir = testDir
+		cmd.Env = envWith(t, "GO111MODULE=off")
```

Comment:
i think this could be simplified to:
```
cmd.Env = append(os.Environ(), "GO11MODULE=off")
```

since duplicate entries causes the last value to be used;
>     // If Env contains duplicate environment keys, only the last
>     // value in the slice for each duplicate key is used.
https://golang.org/pkg/os/exec/#Cmd

Reply:
🤦‍♀